yjit/src/codegen.rs

```diff
@@ -4354,6 +4354,95 @@ fn jit_rb_int_equal(
     true
 }
 
+fn jit_rb_int_mul(
+    jit: &mut JITState,
+    ctx: &mut Context,
+    asm: &mut Assembler,
+    ocb: &mut OutlinedCb,
+    _ci: *const rb_callinfo,
+    _cme: *const rb_callable_method_entry_t,
+    _block: Option<IseqPtr>,
+    _argc: i32,
+    _known_recv_class: *const VALUE,
+) -> bool {
+    if ctx.two_fixnums_on_stack(jit) != Some(true) {
```

Comment:
AFAIK `two_fixnums_on_stack` will return `None` if the status is unknown, so this is checking that we must definitely know that we have two fixnums on stack, and disallows the case where we don't know ?

Reply:
nm I see that it does use peeking.